### ISSUE TITLE
[User flows and navigating] Link or button is not working as expected or did not navigate to the intended destination. (03.11.1)

### ISSUE THREAD

**Issue by shiragoodman** [department-of-veterans-affairs/va.gov-team#63469]
#### VFS team name
Public Websites

#### VFS product name
Income Limits

#### Point of Contact/Reviewers
Erin White - @erinrwhite - IA

*For more information on how to interpret this ticket, please refer to the [Anatomy of a Staging Review issue ticket](https://depo-platform-documentation.scrollhelp.site/collaboration-cycle/Anatomy-of-a-Staging-Review-Issue-ticket.2060320997.html) guidance on Platform Website.

---

### Platform Issue
Link or button is not working as expected or did not navigate to the intended destination.

### Issue Details
The results page contains several links to more information or action links, including a prominent link to "Learn more about income limits and deductions" above the accordions. Once users follow these links and leave the results page, they cannot return to the results page using the browser's back button. Instead they're returned to the introduction screen for the form. Because this issue is not part of the form flow itself it is not a launch blocking issue. 

### VA.gov Experience Standard
[Category Number 03, Issue Number 11](https://depo-platform-documentation.scrollhelp.site/collaboration-cycle/VA.gov-experience-standards.1683980311.html)

### Platform Recommendation
If possible, maintain the state of the form results page so that it can be accessed using the browser back button. Another option to keep users in the flow is to open the "Learn more about income limits and deductions" link in a new window. 

---

### VFS Guidance

- Close the ticket when the issue has been resolved or validated by your Product Owner
- If your team has additional questions or needs Platform help validating the issue, please comment on the ticket
- Some feedback provided may be out of scope for your iteration of the product, however, Platform's OCTO leadership has stated that all identified issues need to be documented and it is still your responsibility to resolve the issue.
- If you do not believe that this Staging Review issue ticket is the responsibility of your team, comment below providing an explanation and who you believe is responsible. Please tag the Point of Contact/Reviewers. Governance team will research and will follow up.

**Comment by wesrowe** [comment]
@randimays, I let this one slip through the cracks last week. I don't know if we discussed the implementation details of the suggestion. I would phrase the question as: "If a Veteran clicks away from the Results screen (e.g., using one of the more-info links on the results), would it be possible in our React/Redux framework for the Results screen to still be populated if they return to it using their browser's back button?"

Our fallback would be to open these links in a new tab in order to preserve the populated Results screen.

**Comment by randimays** [comment]
@wesrowe I think this would fall in the exception category for this DST rule:

> Open in the same window except in certain instances. Links should only open in a new tab if clicking the link will result in the user losing progress or data. Otherwise, links should open in the same window.

Allowing the back button in this scenario would be very tricky if possible at all. Because the Veteran is lead to a new screen, the local session for Income Limits is wiped out. When they try to navigate backward, there is no data in the store to check if they should be there, and the page is routed back to the introduction.

If it's ok with all involved, I think a new tab would be best.

**Comment by wesrowe** [comment]
Sounds good. The new-tab fallback scenario was okayed during the staging review meeting.

**Comment by randimays** [comment]
Is this launch-blocking? If so, maybe @chriskim2311 could pick up?

**Comment by wesrowe** [comment]
@erinrwhite, the ticket specifies the links under "Learn more about income limits and deductions". I think that other informational links should probably open that way also, e.g. "Learn more about income limits and deductions." We'll do that.

I wanted to ask, though: I'm not sure about whether to open the "Apply for VA health care" action links in a new tab. Is there a pattern for this? In theory the user is taking our preferred step upon wizard completion, so wouldn't need the Results screen any more. However, I could imagine someone clicking this just to see where it goes, and then wanting to go back to the Results.

cc @thejordanwood

**Comment by wesrowe** [comment]
also cc @aklausmeier, as this will come up on future projects like the PACT Act wizard.

**Comment by erinrwhite** [comment]
@randimays Not launch-blocking 👍🏻 though now I'm thinking about how the only indication for non-launch-blocking issues is the lack of a `launch-blocking` label, so I'm following up with my team about our ticket format.
 
@wesrowe This is a great question. I don't know of official guidance for this beyond the really helpful [language that Randi found](https://github.com/department-of-veterans-affairs/va.gov-team/issues/63469#issuecomment-1684256055). I chatted with @briandeconinck a little on it. We landed on recommending that those links open in the same tab, since it's an action, a primary flow and the desired exit from the app. But/and we are going to keep surfacing this conversation, especially if it impacts the PACT Act wizard, which will likely have more steps than this one.

**Comment by randimays** [comment]
@erinrwhite Re:

> We landed on recommending that those links open in the same tab, since it's an action, a primary flow and the desired exit from the app.

Does this mean **all** links (more information and action links) should open in the same tab? Or was this comment specific to the `Apply for VA health care` links?

**Comment by erinrwhite** [comment]
@randimays thank you for clarifying! Was thinking it's just for the "Apply for VA health care" action links.

**Comment by wesrowe** [comment]
This feedback was addressed through Public WEbsites ticket https://github.com/department-of-veterans-affairs/va.gov-cms/issues/14858. Closing!
